PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:8 + add:0, change matches:
-    _inherit = "account.journal"
-    _inherit = 'account.move'
-    _inherit = 'account.payment'
-    _inherit = 'account.payment.method'
-    _inherit = "account.payment.method.line"
-    _inherit = 'onboarding.onboarding.step'
-    _inherit = 'payment.provider'
-    _inherit = 'payment.transaction'

Total Changes: 18

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/account_payment/models/account_journal.py
+++ b/addons/account_payment/models/account_journal.py
@@ -6,7 +6,7 @@ from odoo.exceptions import UserError
 
 
 class AccountJournal(models.Model):
-    _inherit = "account.journal"
+    _inherit = ["account.journal"]
 
     def _get_available_payment_method_lines(self, payment_type):
         lines = super()._get_available_payment_method_lines(payment_type)

--- a/addons/account_payment/models/account_move.py
+++ b/addons/account_payment/models/account_move.py
@@ -7,7 +7,7 @@ from odoo.addons.payment import utils as payment_utils
 
 
 class AccountMove(models.Model):
-    _inherit = 'account.move'
+    _inherit = ['account.move']
 
     transaction_ids = fields.Many2many(
         string="Transactions", comodel_name='payment.transaction',

--- a/addons/account_payment/models/account_payment.py
+++ b/addons/account_payment/models/account_payment.py
@@ -5,7 +5,7 @@ from odoo.exceptions import ValidationError
 
 
 class AccountPayment(models.Model):
-    _inherit = 'account.payment'
+    _inherit = ['account.payment']
 
     # == Business fields ==
     payment_transaction_id = fields.Many2one(

--- a/addons/account_payment/models/account_payment_method.py
+++ b/addons/account_payment/models/account_payment_method.py
@@ -5,7 +5,7 @@ from odoo import api, models
 
 
 class AccountPaymentMethod(models.Model):
-    _inherit = 'account.payment.method'
+    _inherit = ['account.payment.method']
 
     @api.model
     def _get_payment_method_information(self):

--- a/addons/account_payment/models/account_payment_method_line.py
+++ b/addons/account_payment/models/account_payment_method_line.py
@@ -7,7 +7,7 @@ from odoo.osv import expression
 
 
 class AccountPaymentMethodLine(models.Model):
-    _inherit = "account.payment.method.line"
+    _inherit = ["account.payment.method.line"]
 
     payment_provider_id = fields.Many2one(
         comodel_name='payment.provider',

--- a/addons/account_payment/models/onboarding_onboarding_step.py
+++ b/addons/account_payment/models/onboarding_onboarding_step.py
@@ -3,8 +3,8 @@
 from odoo import api, models
 
 
-class OnboardingStep(models.Model):
-    _inherit = 'onboarding.onboarding.step'
+class OnboardingOnboardingStep(models.Model):
+    _inherit = ['onboarding.onboarding.step']
 
     @api.model
     def action_open_step_payment_provider(self):

--- a/addons/account_payment/models/payment_provider.py
+++ b/addons/account_payment/models/payment_provider.py
@@ -5,7 +5,7 @@ from odoo.exceptions import UserError
 
 
 class PaymentProvider(models.Model):
-    _inherit = 'payment.provider'
+    _inherit = ['payment.provider']
 
     journal_id = fields.Many2one(
         string="Payment Journal",

--- a/addons/account_payment/models/payment_transaction.py
+++ b/addons/account_payment/models/payment_transaction.py
@@ -4,7 +4,7 @@ from odoo import api, fields, models, SUPERUSER_ID, _
 
 
 class PaymentTransaction(models.Model):
-    _inherit = 'payment.transaction'
+    _inherit = ['payment.transaction']
 
     payment_id = fields.Many2one(
         string="Payment", comodel_name='account.payment', readonly=True)
